lua/ide/components/bufferlist/component.lua

```diff
@@ -93,6 +96,14 @@ BufferListComponent.new = function(name, config)
             end,
             group = vim.api.nvim_create_augroup("NvimIdeBufferlist", { clear = true }),
         })
+        vim.api.nvim_create_autocmd({ "BufEnter" }, {
+            callback = function(args)
+                if libbuf.is_regular_buffer(args.buf) then
+                    vim.schedule(self.refresh)
+                end
+            end,
+            group = vim.api.nvim_create_augroup("NvimIdeBufferlist", { clear = true }),
```

Comment:
This is going to clear out the existing autocmds. If we're adding new autocmds as a separate API call like this, then the `vim.api.nvim_create_augroup` will need to be extracted and put above all autocmds, rather than inline like this, otherwise we're clearing the ones we just created.